common/logger/src/tracing_adapter.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate as dl;
+use std::fmt;
+use tracing as tr;
+use tracing_subscriber::{layer::Context, Layer};
+
+/// A layer that translates tracing events into diem-logger events.
+pub struct TracingToDiemLoggerLayer;
+
+fn translate_level(level: &tr::Level) -> Option<dl::Level> {
+    if *level == tr::Level::ERROR {
+        return Some(dl::Level::Error);
+    }
+    if *level == tr::Level::INFO {
+        return Some(dl::Level::Info);
+    }
+    if *level == tr::Level::DEBUG {
+        return Some(dl::Level::Debug);
+    }
+    if *level == tr::Level::TRACE {
+        return Some(dl::Level::Trace);
+    }
+    if *level == tr::Level::WARN {
+        return Some(dl::Level::Warn);
+    }
+    None
+}
+
+fn translate_metadata(metadata: &tr::Metadata<'static>) -> Option<dl::Metadata> {
+    let level = translate_level(metadata.level())?;
+
+    Some(dl::Metadata::new(
+        level,
+        metadata.target(),
+        metadata.module_path().unwrap_or(""),
+        metadata.file().unwrap_or(""),
+        metadata.line().unwrap_or(0),
+        "", // REVISIT: how to construct a "file:line" with static lifetime here?
```

Comment:
Yeah I suppose our Metadata definition is probably a little too strict

Reply:
I guess this means I should just keep it as "" for now?

Reply:
yeah leave as "", it wouldn't be good to leak